daemon/clone_request.go

```diff
@@ -38,7 +38,10 @@ func CloneExistingRequest(request CloneRequest) *http.Request {
 	newURL = ReconstructURL(request.Request, request.Protocol, request.Host, request.BasePath, request.Port)
 
 	// create cloned request
-	newReq, _ = http.NewRequest(request.Request.Method, newURL, io.NopCloser(bytes.NewBuffer(b)))
+	newReq, err := http.NewRequest(request.Request.Method, newURL, io.NopCloser(bytes.NewBuffer(b)))
+	if err != nil {
+		return nil, fmt.Errorf("error cloning request: %w", err)
```

Comment:
This should not happen, failing at this point is due to something not being correctly handled lower in the stack.

Reply:
In mock mode, when the url flag is missing, the request fails to parse the `newURL` properly because the hostname and scheme is missing.

But, based on the feedback provided, it seems that I should rather handle this issue in another manner.

<img width="1330" alt="image" src="https://github.com/pb33f/wiretap/assets/1235638/b1e46a50-b934-4faa-ac06-0f010eb5f67a">